util/train_test_functions.R

```diff
@@ -376,14 +378,41 @@ PredictWrapper <- function(train.model.list, pred.list, sample.df,
     names(norm.list[[nrm.it]]) <- model.names
   }
 
-  # if returning Kappa -- melt the list into a data.frame and return
+  # if only returning Kappa -- melt the list into a data.frame and return
   # the data.frame
-  if (return.kap) {
+  if (only.kap) {
     kappa.df <- reshape2::melt(norm.list)
     colnames(kappa.df) <- c("kappa", "perc.seq", "classifier", "norm.method")
     return(kappa.df)
-  } else {  # otherwise, return the list of confusionMatrix objects
-    return(norm.list)
+  } else {  # otherwise, return two objects:
+    # 1. the list of confusionMatrix objects (norm.list)
+    # 2. kappa data frame (kappa.df)
+
+    # create kappa.df from norm.list
+    norm_meth_vector <- c()
+    pred_meth_vector <- c()
+    seq_pct_vector <- c()
+    kappa_vector <- c()
+    for(norm_meth in names(norm.list)){
+      for(pred_meth in names(norm.list[[norm_meth]])){
+        for(seq_pct in names(norm.list[[norm_meth]][[pred_meth]])){
+          kappa <- as.numeric(norm.list[[norm_meth]][[pred_meth]][[seq_pct]][["overall"]][["Kappa"]])
+          norm_meth_vector <- c(norm_meth_vector, norm_meth)
+          pred_meth_vector <- c(pred_meth_vector, pred_meth)
+          seq_pct_vector <- c(seq_pct_vector, seq_pct)
+          kappa_vector <- c(kappa_vector, kappa)
+
+        }
+      }
+    }
+    kappa.df <- data.frame(kappa = kappa_vector,
+                           perc.seq = seq_pct_vector,
+                           classifier = pred_meth_vector,
+                           norm.method = norm_meth_vector)
```

Comment:
Hmmmm.... since you called it out in your initial comment - can you use some `purrr` functionality here to pull out the `Kappa` elements (maybe with some filtering functionality) and then flatten (or melt) it into a data.frame?

Reply:
Leaving a note here for reference

https://purrr.tidyverse.org/reference/modify.html
modify_depth()

Reply:
@envest do you plan on making this change before I review this?

Reply:
Yes -- will be on my plan for today

Reply:
☝️ this has now been changed @jaclyn-taroni